### PR TITLE
Sofie tracking

### PR DESCRIPTION
**Pull request by hxie29**
for convenience of documenting remarks

### CODE REVIEW

File: form_validation/built-in_valid/signup.html [hunk 1]
```diff
@@ -26,21 +27,130 @@
...
+        </div>
+
         <!-- section 1 -->
         <div class="container">
             <div class="box">
                 <div class="slogan">
                     <h2>Sign Up</h2><br>
                 </div>
-                <form class="form" id="signup">
+                <form class="form" id="signup" onsubmit="showDiv(event)">
```
Comment (haldax) [inline form_validation/built-in_valid/signup.html:145]: event can be omitted?

File: script/tracking.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
+var clicks = 0;
```
Comment (hxie29) [inline script/tracking.js]: var is deprecated, change to 'let'

File: script/tracking.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+var sexC = 0;
+var languageK = 0;
+var languageC = 0;
+var aboutK = 0;
+var aboutC = 0;
+
+//track the mouse clicks
+document.onclick = function(){
+    clicks++;
+    document.getElementById("track-clicks").innerHTML="Number of mouse clicks: " + clicks;
```
Comment (hxie29) [inline script/tracking.js:35]: it seems that changing the text contents (of all elements) can happen all together once when submit is clicked? (i.e. in showDiv function)

File: script/tracking.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        }
+    }
+}
+
+//make the hidden div visible
+function showDiv(e){
+    e.preventDefault();
+    totalTime();
+    totalCharacters();
+    document.getElementById("hidden-div").style.display="block";
```
Comment (hxie29) [inline script/tracking.js:146]: we can write separate function which changes all innerHTML of elements fields, and invoke that function before changing the display to block

File: script/tracking.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+//make the hidden div with the table visible
+function showFields(){
+    buttonClicks++;         //to determine if the table should be visible
+
+    if((buttonClicks % 2) == 0){
+        document.getElementById("hidden-div-fields").style.display = "none";
+    } else{
+        document.getElementById("userK").innerHTML = userK;
```
Comment (hxie29) [inline script/tracking.js:156]: same comment as above, changing  innerHTML will be done once after click join (submit), in that way we separate the responsibilities, and this function will only have lines about changing the display mode

File: script/tracking.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+        aboutC = about.length;
+    }
+
+    var totalChar = userC + emailC + pwC + fnameC + lnameC + addressC + countryC + zipcodeC + aboutC;
+    document.getElementById("track-total-chars").innerHTML = "Total number of characters typed: " + totalChar;
+}
+
+//count the number of key presses per field
+function addKey(id){
+    document.getElementById(id).onkeyup = function(){
```
Comment (hxie29) [inline script/tracking.js:104]: for this part, I don't know if there is a better way, I will maybe ask TA on wednesday!

File: form_validation/built-in_valid/signup.html [hunk 7]
```diff
@@ -26,21 +27,130 @@
             </div>
         </header>
 
+        <!-- hidden div with tracking stats -->
+        <div id="hidden-div" style="display: none;" class="info_box">
+            <h2>Behavioral data</h2>
+            <p id="track-clicks">Number of mouse clicks: 0</p>
```
Comment (hxie29) [inline form_validation/built-in_valid/signup.html]: i think there is a way to put a placeholder in (so it updates automatically, something with template engine) but I have not learnt it.... :(

Comment (hxie29) [reply form_validation/built-in_valid/signup.html]: I will try learn it on Tuesday, if I figure it out, I will write an update

**Pull request merged**